packages/database.ol

```diff
@@ -33,6 +33,25 @@ type ConnectionInfo:void {
 	.checkConnection?:int // if true (> 0) check connection before each DB command (default: false (0))
 	.toLowerCase?: bool // lowercase attribute names
 	.toUpperCase?: bool // uppercase attribute names
+	.connectionPoolConfig?: ConnectionPoolConfig
+}
+
+// See https://github.com/brettwooldridge/HikariCP?tab=readme-ov-file#frequently-used for descriptions
+type ConnectionPoolConfig: void {
```

Comment:
I left out the following configuration options on purpose:

- autoCommit?: bool	<-- The way the DatabaseService is currently implemented, providing this would simply mean that 'update' and 'query' breaks outside using transactions.
- metricRegistry	<-- Needs to be passed an object
- healthCheckRegistry <-- Needs to be passed an object
- allowPoolSuspension?: bool	<-- The logic for pool suspension is not implemented in the DatabaseService, so we're not allowing this for now
- registerMbeans <-- MBeans requires the user to modify the DatabaseService code, which is not intended
- driverClassName?: string 	<-- This is entered in the ConnectionInfo already, so just resolve it from there
- leakDetectionThreshold?: int <-- Logging is currently not enabled for DatabaseService, so this would be pointless
- dataSource?: DataSource <-- Requires to send a object of type DataSource
- threadFactory?: ThreadFactory <-- Requires user to provide an object of type ThreadFactory
- scheduledExecutor?: ScheduledExecutor <-- Requires user to provide object o f type ScheduledExecutor
- leakDetectionThreshold: int <- Since we are not logginng any information, this is pointless